src/transformers/training_args.py

```diff
@@ -809,6 +826,14 @@ def __post_init__(self):
                 )
             if not (self.sharded_ddp == "" or not self.sharded_ddp):
                 raise ValueError("sharded_ddp is not supported with bf16")
+
+        if self.adafactor:
+            warnings.warn(
```

Comment:
we use `logger.warning` for most warnings so let's swap it.

Reply:
1. Are you sure about `logger.warning`? I followed the other examples in `training_args.py`:

https://github.com/manuelciosici/transformers/blob/7e62da916f392670e8e7151ed25b5cbdcb8b5a92/src/transformers/training_args.py#L759-L763

https://github.com/manuelciosici/transformers/blob/7e62da916f392670e8e7151ed25b5cbdcb8b5a92/src/transformers/training_args.py#L810-L814

https://github.com/manuelciosici/transformers/blob/7e62da916f392670e8e7151ed25b5cbdcb8b5a92/src/transformers/training_args.py#L895-L899

There are a few others.

2. If yes to 1, should I also change the other calls to `warnings.warn` to `logger.warning`?

Reply:
No, you're right. There are only 2 logger.warnings for deprecated API, so it's those then should be fixed for consistency, but in a different PR.

Thank you for analyzing that file for patterns, @manuelciosici